lib/common_changes.ex

```diff
@@ -119,15 +117,20 @@ defmodule EctoShorts.CommonChanges do
   CommonChanges.put_or_cast_assoc(change(user, fruits: [%{id: 1}, %{id: 3}]), :fruits)
   ```
   """
+  @spec put_or_cast_assoc(Changeset.t, atom) :: Changeset.t
+  @spec put_or_cast_assoc(Changeset.t, atom, Keyword.t) :: Changeset.t
   def put_or_cast_assoc(changeset, key, opts \\ []) do
     params_data = Map.get(changeset.params, Atom.to_string(key))
 
     find_method_and_put_or_cast(changeset, key, params_data, opts)
   end
 
+  defp find_method_and_put_or_cast(changeset, key, params_data, opts) when params_data === nil do
```

Comment:
Instead of nil comparisons we should either use is_nil or directly have the arg in the function head as nil to pattern match! Usually the latter is preferable

Reply:
Oopsies! Fixed :)